solr/core/src/java/org/apache/solr/core/FileSystemConfigSetService.java

```diff
@@ -287,9 +287,11 @@ public Long getCurrentSchemaModificationVersion(
   }
 
   protected Path getConfigDir(String configName) throws IOException {
-    String configSetDirPath = configSetBase.toFile().getCanonicalPath() + File.separator;
-    String path = Paths.get(configSetBase.toString(), configName).toFile().getCanonicalPath();
-    if (!path.startsWith(configSetDirPath)) {
+    String path =
+        Paths.get(configSetBase.toString(), FilenameUtils.getName(configName))
+            .toFile()
+            .getCanonicalPath();
+    if (!path.endsWith(configName)) {
```

Comment:
Don't we want to check that the path *startsWith* the configSetBase?  I don't think we should care what it ends with.  Otherwise we could read a path like "../../../somewhereElse/configName" that the caller shouldn't have access to.